packages/sdk/src/instant-trade/InstantTradeBuyerTxBuilder.ts

```diff
@@ -0,0 +1,199 @@
+      {
+        standardOutput: this.sellerPSBT.data.outputs[0],
+        txOutput: (this.sellerPSBT.data.globalMap.unsignedTx as any).tx.outs[0],
+        sats: (this.sellerPSBT.data.globalMap.unsignedTx as any).tx.outs[0].value,
+        injectionIndex: INSTANT_BUY_SELLER_INPUT_INDEX
+      }
+    ] as InjectableOutput[]
+  }
+
+  private async findUTXOs() {
+    const utxos = (
+      await OrditApi.fetchUnspentUTXOs({
+        address: this.address,
+        network: this.network,
+        sort: "asc" // sort by ascending order to use low amount utxos as refundable utxos
+      })
+    ).spendableUTXOs.filter((utxo) => utxo.sats >= MINIMUM_AMOUNT_IN_SATS)
+
+    // 3 = 2 refundables + (at least) 1 to cover for purchase
+    if (utxos.length < 3) {
+      throw new Error("No suitable UTXOs found")
+    }
+
+    // bind minimum utxos. PSBTBuilder will add more if needed
+    return utxos.slice(0, 3)
+  }
+
+  async isEligible() {
+    const [utxos] = await Promise.all([this.findUTXOs(), this.verifyAndFindInscriptionUTXO(this.sellerAddress)])
+    const sortedUTXOs = utxos.sort((a, b) => a.sats - b.sats)
+    const [refundableUTXOOne, refundableUTXOTwo, ...restUTXOs] = sortedUTXOs
+    const refundables = [refundableUTXOOne, refundableUTXOTwo].reduce((acc, curr) => (acc += curr.sats), 0)
+    const spendables = restUTXOs.reduce((acc, curr) => (acc += curr.sats), 0)
+    const eligible = refundables >= MINIMUM_AMOUNT_IN_SATS * 2 && spendables >= MINIMUM_AMOUNT_IN_SATS
+
+    if (eligible) {
+      this.utxos = utxos
+    }
+
+    return {
+      eligible,
+      refundables,
+      spendables
+    }
+  }
+
+  async build() {
+    const { eligible } = await this.isEligible()
+    if (!eligible) {
+      throw new Error("Not eligible")
+    }
+
+    this.decodePrice()
+    this.bindRefundableOutput()
+    this.bindInscriptionOutput()
+    this.mergePSBTs()
+
+    await this.prepare()
+  }
+
+  async splitUTXOsForTrade(destinationAddress: string) {
+    const { totalUTXOs, spendableUTXOs } = await OrditApi.fetchUnspentUTXOs({
+      address: this.address,
+      network: this.network
+    })
+    if (!totalUTXOs) {
+      throw new Error("No UTXOs found")
+    }
+
+    const utxo = spendableUTXOs.sort((a, b) => b.sats - a.sats)[0] // Largest UTXO
+    const input = await processInput({ utxo, pubKey: this.publicKey, network: this.network })
+    const totalOutputs = 3
+    const outputs: Output[] = []
+    this.inputs = [input]
+
+    for (let i = 0; i < totalOutputs; i++) {
+      const usedAmount = outputs.reduce((acc, curr) => (acc += curr.value), 0)
+      const remainingAmount = utxo.sats - usedAmount
+      const amount = [0, 1].includes(i) ? MINIMUM_AMOUNT_IN_SATS : remainingAmount
+
+      if (amount < MINIMUM_AMOUNT_IN_SATS) {
+        throw new Error(
+          `Not enough sats to generate ${totalOutputs} UTXOs with at least ${MINIMUM_AMOUNT_IN_SATS} sats per UTXO. Try decreasing the count or deposit more BTC`
+        )
+      }
```

Comment:
`const totalOutputs = 3` is not a good assumption. In the case the user's largest UTXO is `1500`, and assuming `MINIMUM_AMOUNT_IN_SATS` is `600`, then output 1 will have `600` sats, output 2 will have `600` sats and output 3 will have `300` which will result in error.